packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -491,7 +491,8 @@ class AndroidDevice extends Device {
       return LaunchResult.failed();
     }
 
-    final TargetPlatform devicePlatform = await targetPlatform;
+    final BuildInfo buildInfo = debuggingOptions.buildInfo;
```

Comment:
there is an existing android_device_test file, this needs a unit test that the targetPlatform from build info is used, or the default target platform is used if null